packages/react-aria-components/docs/ListBox.mdx

```diff
@@ -580,6 +590,947 @@ to the `items` prop. See [Collections](../react-stately/Collections.html) for mo
+
+```tsx example export=true
+function DraggableListBox() {
+  let items = new Map([
+    ['ps', {name: 'Photoshop', style: 'strong'}],
+    ['xd', {name: 'XD', style: 'strong'}],
+    ['id', {name: 'InDesign', style: 'strong'}],
+    ['dw', {name: 'Dreamweaver', style: 'em'}],
+    ['co', {name: 'Connect', style: 'em'}]
+  ]);
+
+  let { dragAndDropHooks } = useDragAndDrop({
+    ///- begin collapse -///
+    renderDragPreview(items) {
+      return (
+        <div className="drag-preview">
+          {items[0]['text/plain']}
+          <span className="badge">{items.length}</span>
+        </div>
+      );
+    },
+    ///- end collapse -///
+    /*- begin highlight -*/
+    getItems(keys) {
+      return [...keys].map(key => {
+        let item = items.get(key);
+        return {
+          'text/plain': item.name,
+          'text/html': `<${item.style}>${item.name}</${item.style}>`,
+          'custom-app-type': JSON.stringify({id: key, ...item})
+        };
+      });
+    },
+    /*- end highlight -*/
+  });
+
+  return (
+    <ListBox aria-label="Draggable list" selectionMode="multiple" items={items} dragAndDropHooks={dragAndDropHooks}>
+      {([id, item]) => <Item id={id} textValue={item.name}><item.style>{item.name}</item.style></Item>}
+    </ListBox>
+  );
+}
+
+<div style={{display: 'flex', gap: 12, flexWrap: 'wrap'}}>
+  <DraggableListBox />
+  {/* see below */}
+  <DroppableListBox />
+</div>
+```
+
+### Dropping on the collection
+
+Dropping on the ListBox as a whole can be enabled using the `onRootDrop` event. When a valid drag hovers over the ListBox, it receives the `isDropTarget` state and can be styled using the `[data-drop-target]` CSS selector.
+
+```tsx example
+function Example() {
+  let [items, setItems] = React.useState([]);
+
+  let { dragAndDropHooks } = useDragAndDrop({
+    /*- begin highlight -*/
+    async onRootDrop(e) {
+      let items = await Promise.all(e.items.map(async (item, i) => {
+        let name = await item.getText('text/plain');
+        return {id: i, name};
+      }));
+      setItems(items);
+    }
+    /*- end highlight -*/
+  });
+
+  return (
+    <div style={{display: 'flex', gap: 12, flexWrap: 'wrap'}}>
+      <DraggableListBox />
+      <ListBox aria-label="Droppable list" items={items} dragAndDropHooks={dragAndDropHooks} renderEmptyState={() => 'Drop items here'}>
+        {item => <Item>{item.name}</Item>}
+      </ListBox>
+    </div>
+  );
+}
+```
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.react-aria-ListBox[data-drop-target] {
+  border-color: var(--highlight-background);
+  box-shadow: 0 0 0 1px var(--highlight-background);
+  background: rgb(from slateblue r g b / 15%);
+}
+```
+
+</details>
+
+### Dropping on items
```

Comment:
This example seems has some weirdness with double focus rings once you perform a keyboard drop on an item.

Reply:
Possibly related to the fix for sticky focus rings in https://github.com/adobe/react-spectrum/pull/4221